eras/conway/impl/src/Cardano/Ledger/Conway/Delegation.hs

```diff
@@ -97,42 +88,34 @@ data Delegatee c
   | DelegStakeVote !(KeyHash 'StakePool c) !(Credential 'Voting c)
   deriving (Show, Generic, Eq)
 
-instance Crypto c => DecCBOR (Delegatee c) where
-  decCBOR = decode . Summands "Delegatee" $ \case
-    0 ->
-      SumD DelegStake
-        <! From
-    1 ->
-      SumD DelegVote
-        <! From
-    2 ->
-      SumD DelegStakeVote
-        <! From
-        <! From
-    k -> Invalid k
-
-instance Crypto c => EncCBOR (Delegatee c) where
-  encCBOR x = encode $ case x of
-    DelegStake kh ->
-      Sum DelegStake 0
-        !> To kh
-    DelegVote c ->
-      Sum DelegVote 1
-        !> To c
-    DelegStakeVote kh c ->
-      Sum DelegStakeVote 2
-        !> To kh
-        !> To c
-
 instance NFData (Delegatee c)
 
 instance NoThunks (Delegatee c)
 
+-- | Certificates for registration and delegation of stake to Pools and DReps. Comparing
+-- to previous eras, there is now ability to:
+--
+-- * Register and delegate with a single certificate: `ConwayRegDelegCert`
+--
+-- * Ability to delegate to DReps with `DelegVote` and `DelegStakeVote`
+--
+-- * Ability to specify the deposit amount. Deposits during registration and
+--   unregistration in Conway are optional, which will change in the future era. They are
+--   optional only for the smooth transition from Babbage to Conway. Validity of deposits
+--   is checked by the @CERT@ rule.
 data ConwayDelegCert c
-  = ConwayDeleg !(StakeCredential c) !(Delegatee c) !(StrictMaybe Coin)
-  | ConwayReDeleg !(StakeCredential c) !(Delegatee c)
-  | ConwayUnDeleg !(StakeCredential c) !(StrictMaybe Coin)
-  | ConwayRegKey !(StakeCredential c)
+  = -- | Register staking credential. Deposit, when present, must match the expected deposit
+    -- amount specified by `ppKeyDepositL` in the protocol parameters.
+    ConwayRegCert !(StakeCredential c) !(StrictMaybe Coin)
+  | -- | De-Register the staking credential. Deposit, if present, must match the amount
+    -- that was left as a deposit upon stake credential registration.
+    ConwayUnRegCert !(StakeCredential c) !(StrictMaybe Coin)
+  | -- | Redelegate to another delegatee. Staking credential must already be registered.
+    ConwayDelegCert !(StakeCredential c) !(Delegatee c)
+  | -- | This is a new type of certificate, which allows to register staking credential
+    -- and delegate within a single certificate. Deposit is required and must match the
+    -- expected deposit amount specified by `ppKeyDepositL` in the protocol parameters.
+    ConwayRegDelegCert !(StakeCredential c) !(Delegatee c) !Coin
```

Comment:
These comments are extremely helpful to match with the spec. Thank you! :bow: :heart: